recipes/azure-sdk-for-cpp/all/conanfile.py

```diff
@@ -23,63 +17,96 @@ class AzureSDKForCppConan(ConanFile):
     topics = ("azure", "cpp", "cross-platform", "microsoft", "cloud")
     package_type = "library"
     settings = "os", "arch", "compiler", "build_type"
-    options = {"shared": [True, False], "fPIC": [True, False]}
-    options.update({_name: [True, False] for _name in AZURE_SDK_MODULES})
+
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "win_http_transport": [True, False]
+    }
+
     default_options = {"shared": False, "fPIC": True}
-    default_options.update({_name: True for _name in AZURE_SDK_MODULES}) # Build all modules by default, let users pick what they do not want
+
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "win_http_transport": False
+    }
 
     def export_sources(self):
         copy(self, "CMakeLists.txt", src=self.recipe_folder, dst=self.export_sources_folder)
 
     def source(self):
         get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        else:
+            del self.options.win_http_transport
+
     def configure(self):
         if self.options.get_safe("shared"):
             self.options.rm_safe("fPIC")
+        if self.settings.os != "Windows":
+            self.options.rm_safe("win_http_transport")
 
     def requirements(self):
         self.requires("openssl/[>=1.1 <4]")
-        self.requires("libcurl/[>=7.78 <9]")
         self.requires("libxml2/[>=2.12.5 <3]")
 
+        if self.settings.os == "Windows":
+            # wil is always required on windows since azure-identity can't be disabled via cmake.
+            # azure-identity and wil are not necessary for storage if skip_test=True, but MS
+            # doesn't currently support that build option...
+            self.requires("wil/[>=1.0.231028.1]")
```

Comment:
```suggestion
            self.requires("wil/1.0.250325.1")
```

Please, avoid using version ranges for packages that are not listed https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/dependencies.md#version-ranges

With time, more package should be using version ranges, but they need be reviewed first by the Conan team for compatibility, maintenance, usage, ...